### PR TITLE
Multiple Proposer Slots Allowed Per Epoch for Validators

### LINKED ISSUE #5065: Allow Multiple Proposal Slots in an Epoch
Currently is is assumed that a validator will only have a single proposal slot in an epoch, however due to the fact that proposal slots are computed in on a per slot basis, there is the possibility that a validator could have multiple proposal slots in an epoch instead of once. This issue is to track the implementation of the required changes for this. We will need to modify our ethereum-api and the relevant code in prysm

### PR DESCRIPTION
**Pull request by rauljordan**
Resolves #5065 

---

# Description

**Write why you are making the changes in this pull request**

Given proposer indices are computed on a per-slot basis, there's a possibility a validator could be assigned to proposer twice in an epoch, and we do not yet allow for this functionality.

**Write a summary of the changes you are making**

This PR uses latest changes in EthereumAPIs to modify how we handle duties and assignments for our validators. This PR also deletes the deprecated CommitteeAssignment function which is no longer relevant and instead favors CommitteeAssignments.

### CODE REVIEW

File: beacon-chain/core/helpers/committee.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ import (
 
 var committeeCache = cache.NewCommitteesCache()
 
+type proposerIndexToSlots map[uint64][]uint64
```
Comment (rauljordan) [inline beacon-chain/core/helpers/committee.go:21]:
```suggestion
```

File: beacon-chain/core/helpers/committee.go [hunk 2]
```diff
@@ -18,6 +18,8 @@ import (
...
+
```
Comment (rauljordan) [inline beacon-chain/core/helpers/committee.go:22]:
```suggestion
```

File: beacon-chain/core/helpers/committee_test.go [hunk 3]
```diff
@@ -4,13 +4,13 @@ import (
...
 	"reflect"
 	"strconv"
-	"strings"
 	"testing"
 
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
 	"github.com/prysmaticlabs/go-bitfield"
-	beaconstate "github.com/prysmaticlabs/prysm/beacon-chain/state"
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+
```
Comment (rauljordan) [inline beacon-chain/core/helpers/committee_test.go:12]:
```suggestion
```

File: beacon-chain/core/helpers/committee_test.go [hunk 4]
```diff
@@ -191,148 +190,6 @@ func TestVerifyBitfieldLength_OK(t *testing.T) {
...
-				t.Errorf("Computed different attesting slot for validator %d", i)
-			}
-			if proposerSlot != proposers[i] {
-				t.Errorf("Computed different proposing slot for validator %d", i)
-			}
-		}
-	}
-}
-
 func TestCommitteeAssignments_CanRetrieve(t *testing.T) {
```
Comment (shayzluf) [inline beacon-chain/core/helpers/committee_test.go:193]: can this test be run with more slots per epoch so we could have a validator that gets more then one slot in an epoch to propose for?

File: beacon-chain/core/helpers/committee.go [hunk 5]
```diff
@@ -235,85 +240,7 @@ func CommitteeAssignments(state *stateTrie.BeaconState, epoch uint64) (map[uint6
...
-//    assert epoch <= next_epoch
-//
-//    start_slot = compute_start_slot_at_epoch(epoch)
-//    for slot in range(start_slot, start_slot + SLOTS_PER_EPOCH):
-//        for index in range(get_committee_count_at_slot(state, Slot(slot))):
-//            committee = get_beacon_committee(state, Slot(slot), CommitteeIndex(index))
-//            if validator_index in committee:
-//                return committee, CommitteeIndex(index), Slot(slot)
-//    return None
-func CommitteeAssignment(
```
Comment (nisdas) [inline beacon-chain/core/helpers/committee.go:271]: why remove this ? dont we need it for a single validator

Comment (rauljordan) [reply beacon-chain/core/helpers/committee.go:271]: Deprecated since several months, it isn't used anywhere in the codebase

**Pull request merged**